mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -983,6 +983,43 @@ def get_metric_history_bulk(self, run_ids, metric_key, max_results):
                 for metric in metrics
             ]
 
+    def get_max_step_for_metric(self, run_id, metric_key):
+        with self.ManagedSessionMaker() as session:
+            max_step = (
+                session.query(func.max(SqlMetric.step))
```

Comment:
What happens if `func.max` matches multiple rows?

Reply:
Seems func.max can handle multiple rows with the same max step

Reply:
Got it!